#### Using the  Wine Quality Dataset, test the claim that Wines with a higher quality rating have a higher median alcohol content than wines with a lower quality rating.

### 1.  Formulating the Hypothesis

### First things first, we want to make sure we formulate a hypothesis at the start of our project. This will make sure our interpretation of the results is correct as well as rigorous

### Null Hypothesis:The median alcohol content is the same for all quality ratings.

### Alternative Hypothesis: Wines with a higher quality rating have a higher median alcohol content than wines with a lower quality rating.



 ### 2. Set a confidence level of 95%
### α = 0.05

### The α value is a threshold we set, by which we say “if the probability of observing a result as extreme or more (p-value) is lower than α, then we reject the Null hypothesis”. 

### Since our α=0.05 (indicating 5% probability), our confidence (1 — α) is 95%.

In [15]:
# Import all the libraries necessary
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

from scipy import stats 

In [14]:
# Loading the CSV file into a DataFrame
wine_df = pd.read_csv('winequality-red.csv', sep=';', header=0)
wine_df.head(10)

fixed acidity  volatile acidity  citric acid  residual sugar  chlorides  \
0            7.4              0.70         0.00             1.9      0.076   
1            7.8              0.88         0.00             2.6      0.098   
2            7.8              0.76         0.04             2.3      0.092   
3           11.2              0.28         0.56             1.9      0.075   
4            7.4              0.70         0.00             1.9      0.076   
5            7.4              0.66         0.00             1.8      0.075   
6            7.9              0.60         0.06             1.6      0.069   
7            7.3              0.65         0.00             1.2      0.065   
8            7.8              0.58         0.02             2.0      0.073   
9            7.5              0.50         0.36             6.1      0.071   

   free sulfur dioxide  total sulfur dioxide  density    pH  sulphates  \
0                 11.0                  34.0   0.9978  3.51       0.56   
1                 25.0                  67.0   0.9968  3.20       0.68   
2                 15.0                  54.0   0.9970  3.26       0.65   
3                 17.0                  60.0   0.9980  3.16       0.58   
4                 11.0                  34.0   0.9978  3.51       0.56   
5                 13.0                  40.0   0.9978  3.51       0.56   
6                 15.0                  59.0   0.9964  3.30       0.46   
7                 15.0                  21.0   0.9946  3.39       0.47   
8                  9.0                  18.0   0.9968  3.36       0.57   
9                 17.0                 102.0   0.9978  3.35       0.80   

   alcohol  quality  
0      9.4        5  
1      9.8        5  
2      9.8        5  
3      9.8        6  
4      9.4        5  
5      9.4        5  
6      9.4        5  
7     10.0        7  
8      9.5        7  
9     10.5        5

In [9]:
# getting the unique items in the quality column

wine_df.quality.unique()

array([5, 6, 7, 4, 8, 3], dtype=int64)

In [10]:
# grouping by quality and finding median of alcohol content

wine_df2 = wine_df.groupby('quality')['alcohol'].median().to_dict()

wine_df2

{3: 9.925, 4: 10.0, 5: 9.7, 6: 10.5, 7: 11.5, 8: 12.15}

In [11]:
# appending alcohol content on lower quality and higher quality list

lower_quality = []
high_quality =  []
for key,value in  wine_df2.items():
    if key <= 5:
        lower_quality.append(value)
    else:
        high_quality.append(value)

In [12]:
#print the results.
print ( 'high quality : ' ,high_quality)
print ( 'lower quality : ' ,lower_quality)

high quality :  [10.5, 11.5, 12.15]
lower quality :  [9.925, 10.0, 9.7]


In [16]:
#  performing a one sample, two sample t test

def two_sample_test(sample_1, sample_2):
    results = stats.ttest_ind(a = sample_1, b = sample_2)

    return results


two_sample_test(lower_quality, high_quality)

Ttest_indResult(statistic=-3.0891690365598996, pvalue=0.03660386137423448)

## Conclusion:
###  Since the P_value 0.0366 is less than 0.05, we reject the null hypothesis.